File: packages/nextjs/unpin.js
const axios = require("axios");
const PINATA_JWT =
  "eyJ1c2VySW5mb3JtYXRpb24iOnsiaWQiOiIyMmYyMGMwOS1kMDE1LTQ2MTEtYmM1YS02NzdkYzY5Zjk5MzMiLCJlbWFpbCI6InNjb2JydTE5ODhAZ21haWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsInBpbl9wb2xpY3kiOnsicmVnaW9ucyI6W3siaWQiOiJGUkExIiwiZGVzaXJlZFJlcGxpY2F0aW9uQ291bnQiOjF9LHsiaWQiOiJOWUMxIiwiZGVzaXJlZFJlcGxpY2F0aW9uQ291bnQiOjF9XSwidmVyc2lvbiI6MX0sIm1mYV9lbmFibGVkIjpmYWxzZSwic3RhdHVzIjoiQUNUSVZFIn0sImF1dGhlbnRpY2F0aW9uVHlwZSI6InNjb3BlZEtleSIsInNjb3BlZEtleUtleSI6ImMwNGFkOGJlYzliM2Q1NWJjNDc5Iiwic2NvcGVkS2V5U2VjcmV0IjoiOTQ3ODRhMzNmYjI3YzBiZTE5YWU0MjM2MmJjMjRlMWFlYTBlNTZkMjVlY2U2NTY1ZTc1ODM4ZjJlNTdlYjlkMSIsImlhdCI6MTY5NTIzOTc4OH0._FwITlqcuVnLkT40tozVXsAyx_tDbYDGTClVv6mT8L4";
const PIN_QUERY = "https://api.pinata.cloud/data/pinList?status=pinned&includesCount=false&pageLimit=1000";

let pinHashes = [];

const deletePinFromIPFS = async hashToUnpin => {
  try {
    const res = await axios.delete(`https://api.pinata.cloud/pinning/unpin/${hashToUnpin}`, {
      headers: {
        Authorization: PINATA_JWT,
      },
    });
    console.log(`Successfully deleted ${hashToUnpin}`);
  } catch (error) {
    console.log("Error deleting pin:", error.response ? error.response.data : error);
  }
};

const wait = async time => new Promise(resolve => setTimeout(resolve, time));


const fetchPins = async () => {
  try {
    const res = await axios.get(PIN_QUERY, {
      headers: {
        Authorization: PINATA_JWT
      }
    })
    const responseData = res.data.rows
    responseData.forEach(row => {
      pinHashes.push(row.ipfs_pin_hash)
    })
    console.log(pinHashes)
  } catch (error) {
    console.log(error)
  }
}


const bulkUnpin = async () => {
  try {
    for (const hash of pinHashes) {
      await deletePinFromIPFS(hash)
      await wait(200)
    }
    pinHashes = []
  } catch (error) {
    console.log(error)
  }
}

const main = async () => {
  await fetchPins()
  while (pinHashes) {
    await bulkUnpin()
    await fetchPins()
  }
}

main();

File: packages/nextjs/unpin.py
import requests
import json

# Your Pinata API Key and Secret API Key
API_KEY = "e5623b637e18e5fe8836"
SECRET_API_KEY = "2d535984c3bfde9767bf765287669747e3beeafd922863d09e93052287213372"

# Base API URL for Pinata
BASE_URL = "https://api.pinata.cloud/"

# Headers
HEADERS = {
    "pinata_api_key": API_KEY,
    "pinata_secret_api_key": SECRET_API_KEY,
}


def fetch_all_pins():
    url = f"{BASE_URL}data/pinList"
    response = requests.get(url, headers=HEADERS)
    
    if response.status_code != 200:
        print(f"Failed to fetch pinned items: {response.json()}")
        return []
    
    return response.json()["rows"]

def unpin_and_delete_content(hash_to_unpin):
    url = f"{BASE_URL}pinning/pinByHash/{hash_to_unpin}"
    params = {'deleteWhenDone': 'true'}
    response = requests.delete(url, headers=HEADERS, params=params)
    
    if response.status_code == 200:
        print(f"Successfully unpinned and deleted {hash_to_unpin}")
    else:
        print(f"Failed to unpin and delete {hash_to_unpin}: {response.json()}")


def unpin_and_delete_content(hash_to_unpin):
    url = f"{BASE_URL}pinning/pinByHash/{hash_to_unpin}"
    params = {'deleteWhenDone': 'true'}
    response = requests.delete(url, headers=HEADERS, params=params)
    
    if response.status_code == 200:
        print(f"Successfully unpinned and deleted {hash_to_unpin}")
    else:
        print(f"Failed to unpin and delete {hash_to_unpin}: {response.json()}")



if __name__ == "__main__":
   # Fetch all pinned content
    all_pinned_content = fetch_all_pins()

    # Unpin and delete each content
    for content in all_pinned_content:
        ipfs_hash = content["ipfs_pin_hash"]
        unpin_and_delete_content(ipfs_hash)
